core/lib/spree/core/search/base.rb

```diff
@@ -89,7 +89,7 @@ def prepare(params)
 
             per_page = params[:per_page].to_i
             @properties[:per_page] = per_page > 0 ? per_page : Spree::Config[:products_per_page]
-            @properties[:page] = (params[:page].to_i <= 0) ? 1 : params[:page].to_i
+            @properties[:page] = !params[:page].respond_to?(:to_i) ? 0 : ((params[:page].to_i <= 0) ? 1 : params[:page].to_i)
```

Comment:
Line is too long. [125/100]<br>Ternary operators must not be nested. Prefer `if`/`else` constructs instead.